tests/unit/logic/vote.spec.js

```diff
@@ -0,0 +1,1012 @@
+				trs_fail_8 = {asset: {votes: [123, 'abc']}}
+				schema = new zSchema()
+				vote = new Vote(logger, schema)
+			})
+
+			context('Fail 1: if trs.asset is undefined', function () {
+				it('should throw an exception', function () {
+					execMethod = function () { vote.objectNormalize(trs_fail_1) }
+					expect(execMethod).to.throw('Expected type object but found type undefined')
+				})
+			})
+
+			context('Fail 2: if trs.asset.votes is undefined', function () {
+				it('should throw an exception', function () {
+					execMethod = function () { vote.objectNormalize(trs_fail_2) }
+					expect(execMethod).to.throw('Missing required property: votes')
+				})
+			})
+
+			context('Fail 3: if trs.asset.votes is null', function () {
+				it('should throw an exception', function () {
+					execMethod = function () { vote.objectNormalize(trs_fail_3) }
+					expect(execMethod).to.throw('Expected type array but found type null')
+				})
+			})
+
+			context('Fail 4: if trs.asset.votes is an empty Array', function () {
+				it('should throw an exception', function () {
+					execMethod = function () { vote.objectNormalize(trs_fail_4) }
+					expect(execMethod).to.throw('Array is too short')
+				})
+			})
+
+			context('Fail 5: if trs.asset.votes has duplicates', function () {
+				it('should throw an exception', function () {
+					execMethod = function () { vote.objectNormalize(trs_fail_5) }
+					expect(execMethod).to.throw('Array items are not unique')
+				})
+			})
+
+			context('Fail 6: if trs.asset.votes is greater than constants.maxVotesPerTransaction', function () {
+				it('should throw an exception', function () {
+					execMethod = function () { vote.objectNormalize(trs_fail_6) }
+					expect(execMethod).to.throw('Array is too long')
+				})
+			})
+
+			context('Fail 7: if trs.asset has additional properties', function () {
+				it('should throw an exception', function () {
+					execMethod = function () { vote.objectNormalize(trs_fail_7) }
+					expect(execMethod).to.throw('Additional properties not allowed')
+				})
+			})
+
+			context('Fail 8: if trs.asset.votes items has a wrong format', function () {
+				it('should throw an exception', function () {
+					execMethod = function () { vote.objectNormalize(trs_fail_8) }
+					expect(execMethod).to.throw('String does not match pattern')
+					expect(execMethod).to.throw('Expected type string but found type integer')
+				})
+			})
+
+			context('if everything is ok', function () {
+				it('should return the same transaction object', function () {
+					var result = vote.objectNormalize(trs_success)
+					expect(result).to.deep.equal(trs_success)
+				})
+			})
+		})
+
+		describe('dbRead()', function () {
+
+			var raw_success, raw_fail_1, raw_fail_2, raw_fail_3, raw_fail_4, raw_fail_5
+
+			beforeEach(function () {
+				raw_fail_1 = {v_votes: undefined}
+				raw_fail_2 = {v_votes: null}
```

Comment:
Please move this logic inside each individual test..

Reply:
Also, do we really need to test all this cases? At the end the code to test is only `!raw.v_votes` which can be only `true` or `false`

Reply:
Logic moved inside each individual test.

About to test all the cases I think that give us more integrity for to be sure that raw.v_votes have the correct format before to call to raw.v_votes.split()